shell/src/main/java/alluxio/cli/AlluxioShell.java

```diff
@@ -58,9 +58,12 @@
   public static void main(String[] argv) throws IOException {
     int ret;
 
-    if (!Configuration.containsKey(PropertyKey.MASTER_HOSTNAME)) {
-      System.out.println("Cannot run alluxio shell; master hostname is not configured. Please set "
-          + PropertyKey.MASTER_HOSTNAME.toString() + " in alluxio-site.properties.");
+    if (!Configuration.masterHostConfigured()) {
+      System.out.println(String.format(
+          "Cannot run alluxio shell; master hostname is not "
+              + "configured. Please either configure %s in alluxio-site.properties, or configure the "
+              + "shell to look up the master address via Zookeeper",
```

Comment:
They would need to set alluxio.zookeeper.enabled to true and alluxio.zookeeper.address to the addresses of their zookeeper nodes. I omitted the configuration details here to keep the message concise. I think at least 90% of the time when a user gets this error, they aren't using zookeeper.